### PR TITLE
[UX] Add environment variable `SKY_NUM_GPUS_PER_NODE`

### LINKED ISSUE #1334: [UX] An evironment variable indicating the number of GPUs scheduled for the task
A pilot user requests a variable that indicates the number of GPUs scheduled for the task. It can be a variable called `SKY_NUM_GPUS_PER_NODE`.

### PR DESCRIPTION
**Pull request by Michaelvll**
Closes #1334.

Tested:
- [x] The distributed yaml from the doc: `sky launch -c multi-node multi-node.yaml`
```
# multi-node.yaml
name: resnet-distributed-app

resources:
  accelerators: V100:4

num_nodes: 2

setup: |
  pip3 install --upgrade pip
  git clone https://github.com/michaelzhiluo/pytorch-distributed-resnet
  cd pytorch-distributed-resnet && pip3 install -r requirements.txt
  mkdir -p data  && mkdir -p saved_models && cd data && \
    wget -c --quiet https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz
  tar -xvzf cifar-10-python.tar.gz

run: |
  cd pytorch-distributed-resnet

  num_nodes=`echo "$SKY_NODE_IPS" | wc -l`
  master_addr=`echo "$SKY_NODE_IPS" | head -n1`
  python3 -m torch.distributed.launch --nproc_per_node=$SKY_NUM_GPUS_PER_NODE \
    --nnodes=$num_nodes --node_rank=${SKY_NODE_RANK} --master_addr=$master_addr \
    --master_port=8008 resnet_ddp.py --num_epochs 20
```

### CODE REVIEW

File: docs/source/examples/distributed-jobs.rst [hunk 1]
```diff
@@ -47,3 +48,5 @@ SkyPilot exposes two environment variables to distinguish per-node commands:
   You can retrieve the number of nodes by :code:`echo "$SKY_NODE_IPS" | wc -l`
   and the IP address of the third node by :code:`echo "$SKY_NODE_IPS" | sed -n
   3p`.
+- :code:`SKY_NUM_GPUS_PER_NODE`: number of GPUs reserved on each node (the same
```
Comment (concretevitamin) [inline docs/source/examples/distributed-jobs.rst:51]: L41: `SkyPilot exposes these environment variables that can be accessed in a task's ``run`` commands:`

Comment (concretevitamin) [inline docs/source/examples/distributed-jobs.rst:51]: nit: number of GPUs reserved on each node to execute the task; the same as the count in ``accelerators: <name>:<count>``.



nit: L43 add a period?

File: sky/backends/cloud_vm_ray_backend.py [hunk 2]
```diff
@@ -339,17 +339,19 @@ def add_ray_task(self,
         cpu_str = f', num_cpus={backend_utils.DEFAULT_TASK_CPU_DEMAND}'
 
         resources_str = ''
+        num_gpus = 0
         num_gpus_str = ''
         if ray_resources_dict is not None:
             assert len(ray_resources_dict) == 1, \
                 ('There can only be one type of accelerator per instance.'
                  f' Found: {ray_resources_dict}.')
+            num_gpus = list(ray_resources_dict.values())[0]
```
Comment (concretevitamin) [inline sky/backends/cloud_vm_ray_backend.py:348]: What if user does a `sky exec --gpus V100:0.5`?

**Review by concretevitamin — COMMENTED** [review]
Thanks @Michaelvll - a question about fractional GPUs. Should we also add this to a smoke test?

**Comment by Michaelvll** [comment]
Added the smoke test for the new enviornment variable and passed it. PTAL @concretevitamin ;)

File: docs/source/examples/distributed-jobs.rst [hunk 3]
```diff
@@ -38,15 +38,15 @@ For example, here is a simple PyTorch Distributed training example:
...
 - :code:`SKY_NODE_IPS`: a string of IP addresses of the nodes reserved to execute
   the task, where each line contains one IP address.
 
   You can retrieve the number of nodes by :code:`echo "$SKY_NODE_IPS" | wc -l`
   and the IP address of the third node by :code:`echo "$SKY_NODE_IPS" | sed -n
   3p`.
-- :code:`SKY_NUM_GPUS_PER_NODE`: number of GPUs reserved on each node (the same
-  for all nodes scheduled for the task).
+- :code:`SKY_NUM_GPUS_PER_NODE`: number of GPUs reserved on each node to execute the
+  task; rounded to the smallest integer larger than the count in accelerators: <name>:<count>
```
Comment (concretevitamin) [inline docs/source/examples/distributed-jobs.rst:52]:
```suggestion
- :code:`SKY_NUM_GPUS_PER_NODE`: number of GPUs reserved on each node to execute the
  task;  the same as the count in ``accelerators: <name>:<count>`` (rounded up if a fraction).
```

File: tests/test_smoke.py [hunk 4]
```diff
@@ -294,6 +294,9 @@ def test_job_queue():
             f'sky cancel {name} 2',
             'sleep 5',
             f'sky queue {name} | grep {name}-3 | grep RUNNING',
+            f'sky cancel {name} 3',
+            f'sky exec {name} --gpus K80:0.2 "[[ \$SKY_NUM_GPUS_PER_NODE -eq 1 ]] || exit 1"',
```
Comment (concretevitamin) [inline tests/test_smoke.py:298]: Maybe also add a "normal" case, `--gpus K80:1`.

**Review by concretevitamin — APPROVED** [review]
Thanks @Michaelvll, LGTM.

**Pull request merged**